spyder/plugins/ipythonconsole/widgets/main_widget.py

```diff
@@ -1346,15 +1422,17 @@ def get_current_shellwidget(self):
     @Slot(bool)
     @Slot(str)
     @Slot(bool, str)
+    @Slot(bool, str, str)
     @Slot(bool, bool)
     @Slot(bool, str, bool)
     def create_new_client(self, give_focus=True, filename='', is_cython=False,
                           is_pylab=False, is_sympy=False, given_name=None,
-                          cache=True, initial_cwd=None):
+                          cache=True, initial_cwd=None, environment='A',
+                          path_to_custom_interpreter=None):
         """Create a new client"""
         self.master_clients += 1
         client_id = dict(int_id=str(self.master_clients),
-                         str_id='A')
+                         str_id=environment)
```

Comment:
I was checking this and I think the tab name generation should keep the `A` id untouched (that is used to identify multiple consoles using the same kernel) and instead add a new element to the tab name and only with the env name. So for example an env that appears as `conda: my_env (Python 3.9.16)` in the console menu should generate as tab name `Console X (my_env)/A` where only `Console X` is editable text. We should discuss if we should remove or not the `Console` text but that is to be done in the issue (I think it make sense to remove the `Console` text but we need to check).

For the above probably we will need to add a new key to the `clien_id` dict like `str_env_name` which then will be used to create the tab name at: https://github.com/spyder-ide/spyder/blob/dd528385e9bbe589f9e033e59e5f2f33479ee811/spyder/plugins/ipythonconsole/widgets/client.py#L510-L526

There the client `self.id_` attribute will contain the `str_env_name` key

Reply:
@dalthviz, I think that `A` should be removed. I don't see a compelling reason to _distinguish_ between consoles using the same kernel. I see a more compelling case to _identify_ consoles that are using the _same_ kernel. However, currently in Spyder, opening another console will start another kernel, yet both consoles will indicate `/A` implying that both are using the same kernel when they aren't.

It seems to me to be less confusing to just get rid of `A` altogether and in the future determine how to identify consoles that are using the _same_ kernel.

Reply:
Thanks for the comment @mrclary ! We need to discuss that for sure as well as the usage of `Console`. I will add a comment on the issue to track your new comment regarding the `str_id` and the option I described here on how the console name should be formatted :+1: